internal/exec/yaml_func_store_test.go

```diff
@@ -0,0 +1,99 @@
+package exec
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/alicebob/miniredis/v2"
+	"github.com/cloudposse/atmos/pkg/schema"
+	"github.com/cloudposse/atmos/pkg/store"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestProcessTagStore(t *testing.T) {
+	// Start a new Redis server
+	s := miniredis.RunT(t)
+	defer s.Close()
+
+	// Setup the Redis ENV variable
+	redisUrl := fmt.Sprintf("redis://%s", s.Addr())
+	origRedisUrl := os.Getenv("ATMOS_REDIS_URL")
+	os.Setenv("ATMOS_REDIS_URL", redisUrl)
+	defer os.Setenv("ATMOS_REDIS_URL", origRedisUrl)
+
+	// Create a new Redis store
+	redisStore, err := store.NewRedisStore(store.RedisStoreOptions{
+		URL: &redisUrl,
+	})
+	assert.NoError(t, err)
+
+	// Setup test configuration
+	atmosConfig := schema.AtmosConfiguration{
+		Stores: map[string]store.Store{
+			"redis": redisStore,
+		},
+	}
+
+	// Populate the store with some data
+	require.NoError(t, redisStore.Set("dev", "vpc", "cidr", "10.0.0.0/16"))
+	require.NoError(t, redisStore.Set("prod", "vpc", "cidr", "172.16.0.0/16"))
+
+	tests := []struct {
+		name         string
+		input        string
+		currentStack string
+		expected     interface{}
+	}{
+		{
+			name:         "lookup using current stack",
+			input:        "!store redis vpc cidr",
+			currentStack: "dev",
+			expected:     "10.0.0.0/16",
+		},
+		{
+			name:         "basic lookup cross-stack",
+			input:        "!store redis dev vpc cidr",
+			currentStack: "prod",
+			expected:     "10.0.0.0/16",
+		},
+		{
+			name:         "lookup with default value without quotes",
+			input:        "!store redis staging vpc cidr | default 172.20.0.0/16",
+			currentStack: "dev",
+			expected:     "172.20.0.0/16",
+		},
+		{
+			name:         "lookup with default value with single quotes",
+			input:        "!store redis staging vpc cidr | default '172.20.0.0/16'",
+			currentStack: "dev",
+			expected:     "172.20.0.0/16",
+		},
+		{
+			name:         "lookup with default value with double quotes",
+			input:        "!store redis staging vpc cidr | default \"172.20.0.0/16\"",
```

Comment:
note, in YAML, this will look different, if quotes are used.
```
vars:
  foo: !store 'redis staging vpc cidr | default "172.20.0.0/16"'
```

Reply:
```
component3:
      metadata:
        component: hook-and-store
      vars:
        stage: test
        random: !store testredis component-fake random_id | default "foo-with-quotes"

    component4:
      metadata:
        component: hook-and-store
      vars:
        stage: test
        random: !store testredis component-fake random_id | default foo-without-quotes
```

I think I'm missing your point. Both of these examples work fine. Is there something else I should test?

Reply:
I am basing it off of these 2 comments from @aknysh 
- https://github.com/cloudposse/atmos/pull/943#discussion_r1922667038
- https://github.com/cloudposse/atmos/pull/943#discussion_r1922691109

> This is not a valid YAML syntax
> ```yaml
> config: !include "~/My Documents/dev/config.json" "<yq-expression-with-spaces>"
> ```
> the YAML parser does not like when more than one string is defined for the same value

Reply:
I would prefer to be wrong or confused =)

Reply:
Well, I can confirm that the two scenarios above work, so I'm not sure. If there are bugs as a result, we can definitely make any necessary changes.